client/src/components/global/Utility/ScrollLock.js

```diff
@@ -0,0 +1,70 @@
+import React, { PureComponent, PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+
+export default class ScrollLock extends PureComponent {
+
+  static displayName = "Utility.ScrollLock";
+
+  static propTypes = {
+    children: React.PropTypes.element.isRequired
+  };
+
+  constructor() {
+    super();
+
+    this.state = {
+      touchStart: null
+    };
+
+    this.edgeLock = this.edgeLock.bind(this);
+  }
+
+  edgeLock(scroller, delta) {
+    if (delta < 0 && scroller.scrollTop <= 0) {
+      event.preventDefault();
+    }
+
+    if (delta > 0 && scroller.scrollTop >= scroller.scrollHeight - window.innerHeight) {
+      event.preventDefault();
+    }
+  }
+
+  componentDidMount() {
+    this.scroller = typeof this.props.children.type === 'string' ?
```

Comment:
In general, I prefer to use Lodash here for legibility and consistency. Throughout the code, I use

```
import isString from 'lodash/isString';
isString(this.props.children.type) 
```
instead of 
```
typeof this.props.children.type === 'string';
```
The former seems cleaner to me, and that utility function is already widely included.